web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackTypeHelper.php

```diff
@@ -157,6 +166,31 @@ public function validateConfiguration(array $configuration): void {
     }
   }
 
+  /**
+   * Get a specific threshold.
+   */
+  public function getThreshold($termId, $key, ?string $threshold = NULL): int {
+    $storedThresholds = $this->state->get('ai_screening_project_track_thresholds', []);
+    $threshold = $storedThresholds["{$threshold}-{$termId}-{$key}"] ?? 0;
+
+    return $threshold ? (int) $threshold : 0;
+  }
+
+  /**
+   * Get all thresholds as a keyed array.
+   */
+  public function getThresholds(): array {
+    $thresholds = [];
+    $storedThresholds = $this->state->get('ai_screening_project_track_thresholds', []);
+
+    foreach ($storedThresholds as $key => $threshold) {
+      $thresholdKeyArr = explode('-', $key);
+      $thresholds[$thresholdKeyArr[1]][$thresholdKeyArr[2]][$thresholdKeyArr[0]] = $threshold;
```

Comment:
Consider using https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21NestedArray.php/class/NestedArray/11.x to get the value. Or store an array in the store.

Reply:
Changed to array storage